storm/src/main/storm/mesos/util/MesosCommon.java

```diff
@@ -60,7 +60,7 @@
   public static final String SUPERVISOR_ID = "supervisorid";
   public static final String ASSIGNMENT_ID = "assignmentid";
   public static final String DEFAULT_WORKER_NAME_PREFIX_DELIMITER = "_";
-  public static final String DEFAULT_MESOS_COMPONENT_NAME_DELIMITER = " | ";
+  public static final String DEFAULT_MESOS_COMPONENT_NAME_DELIMITER = "|";
```

Comment:
Also, it seems like `DEFAULT_MESOS_COMPONENT_NAME_DELIMITER` & `getMesosComponentNameDelimiter()` weren't used anywhere except tests before your change, right?

It *used* to be used per my changes here:
* https://github.com/mesos/storm/commit/f7f18f86272c1f02e4574996a207e53ab599a8e8

However it seems those changes were destroyed later.  I *definitely* wanna leave the ` | ` styling for the component *names*.  However, for the IDs we might want or need to not have spaces around the delimiter.  I should also reapply my changes that made the name delimiter configurable, instead of hardcoding it to ` | ` as was done later (maybe by me...).

So I think it might be appropriate to use a different field (`MESOS_COMPONENT_ID_DELIMITER`), and have it hardcoded (not configurable) instead of reusing the *name* delimiter for this functionality.